apps/heft/src/plugins/FileGlobSpecifier.ts

```diff
@@ -129,39 +136,55 @@ export async function watchGlobAsync(
   return results;
 }
 
-export async function getFilePathsAsync(
-  fileGlobSpecifier: IFileSelectionSpecifier,
-  fs?: FileSystemAdapter
-): Promise<Set<string>> {
-  const rawFiles: string[] = await glob(fileGlobSpecifier.includeGlobs!, {
-    fs,
+export async function getFileSelectionSpecifierPathsAsync(
+  options: IGetFileSelectionSpecifierPathsOptions
+): Promise<Map<string, fs.Dirent>> {
+  const { fileGlobSpecifier, includeFolders, fileSystemAdapter } = options;
+  const rawEntries: Entry[] = await glob(fileGlobSpecifier.includeGlobs!, {
+    fs: fileSystemAdapter,
     cwd: fileGlobSpecifier.sourcePath,
     ignore: fileGlobSpecifier.excludeGlobs,
     dot: true,
-    absolute: true
+    absolute: true,
+    onlyFiles: !includeFolders,
+    stats: true
```

Comment:
I just know I'm going to have to write my own glob implementation sooner or later so that I can stop wasting CPU cycles making `stat` calls when `fs.readdir` returns everything any decent globber needs to know.

Reply:
Aha! Use `objectMode: true` instead: https://github.com/mrmlnc/fast-glob#objectmode